diff --git a/third_party/devtools-frontend/src/scripts/build/typescript/typescript_vars.gni b/third_party/devtools-frontend/src/scripts/build/typescript/typescript_vars.gni
--- a/third_party/devtools-frontend/src/scripts/build/typescript/typescript_vars.gni
+++ b/third_party/devtools-frontend/src/scripts/build/typescript/typescript_vars.gni
@@ -10,7 +10,7 @@
 } else if (host_os == "win") {
   tsc_binary = "//third_party/typescript/windows-amd64/src/lib/tsc.exe"
 } else if (host_os == "mac") {
-  tsc_binary = "/opt/homebrew/bin/tsc"
+  tsc_binary = devtools_location_prepend + "third_party/typescript/tsc"
 }

 # Standard TypeScript library definitions needed for remote execution (RBE /
diff --git a/third_party/devtools-frontend/src/third_party/typescript/typescript.py b/third_party/devtools-frontend/src/third_party/typescript/typescript.py
--- a/third_party/devtools-frontend/src/third_party/typescript/typescript.py
+++ b/third_party/devtools-frontend/src/third_party/typescript/typescript.py
@@ -10,7 +10,8 @@


 def GetBinaryPath():
-    return "/opt/homebrew/bin/tsc"
+    if platform.system() == 'Darwin':
+        return os_path.join(os_path.dirname(__file__), 'tsc')
     if platform.machine() == 'arm64':
         darwin_path = 'mac-arm64'
     else:
diff --git a/third_party/devtools-frontend/src/scripts/build/esbuild.js b/third_party/devtools-frontend/src/scripts/build/esbuild.js
--- a/third_party/devtools-frontend/src/scripts/build/esbuild.js
+++ b/third_party/devtools-frontend/src/scripts/build/esbuild.js
@@ -19,10 +19,9 @@
 // esbuild module uses binary in this path.
 const binaryName = os.type() === 'Windows_NT' ? 'esbuild.exe' : 'esbuild';
 process.env.ESBUILD_BINARY_PATH = path.join(
-    '/',
-    'opt',
-    'homebrew',
-    'bin',
+    devtoolsRootPath(),
+    'third_party',
+    'esbuild',
     binaryName,
 );
 // This needs to be after the ESBUILD_BINARY_PATH is set
diff --git a/third_party/devtools-frontend/src/scripts/build/ninja/bundle.gni b/third_party/devtools-frontend/src/scripts/build/ninja/bundle.gni
--- a/third_party/devtools-frontend/src/scripts/build/ninja/bundle.gni
+++ b/third_party/devtools-frontend/src/scripts/build/ninja/bundle.gni
@@ -35,7 +35,7 @@
       devtools_location_prepend + "scripts/devtools_paths.js",
     ]

-    _esbuild = "/opt/homebrew/bin/esbuild"
+    _esbuild = devtools_location_prepend + "third_party/esbuild/esbuild"
     if (host_os == "win") {
       inputs += [ _esbuild + ".exe" ]
     } else {
diff --git a/third_party/devtools-frontend/src/scripts/build/typescript/typescript.gni b/third_party/devtools-frontend/src/scripts/build/typescript/typescript.gni
--- a/third_party/devtools-frontend/src/scripts/build/typescript/typescript.gni
+++ b/third_party/devtools-frontend/src/scripts/build/typescript/typescript.gni
@@ -190,7 +190,7 @@

     if (devtools_skip_typecheck) {
       args += [ "--use-esbuild" ]
-      _esbuild = "/opt/homebrew/bin/esbuild"
+      _esbuild = devtools_location_prepend + "third_party/esbuild/esbuild"
       if (host_os == "win") {
         inputs += [ _esbuild + ".exe" ]
       } else {
diff --git a/third_party/devtools-frontend/src/scripts/devtools_paths.py b/third_party/devtools-frontend/src/scripts/devtools_paths.py
--- a/third_party/devtools-frontend/src/scripts/devtools_paths.py
+++ b/third_party/devtools-frontend/src/scripts/devtools_paths.py
@@ -55,7 +55,7 @@


 def esbuild_path():
-    return '/opt/homebrew/bin/esbuild'
+    return path.join(devtools_root_path(), 'third_party', 'esbuild', 'esbuild')


 def license_checker_path():
